### PR TITLE
refactor: `unfold` constructor is not variadic anymore

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- [ ] Fix ...
- [ ] Provide ...
- [ ] It breaks backward compatibility
- [ ] Is covered by unit tests
- [ ] Has static analysis tests (psalm, phpstan)
- [ ] Has documentation
- [ ] Is an experimental thing

Follows #. Related to #. Fixes #.

### CODE REVIEW

File: src/Contract/Operation/Unfoldable.php [hunk 1]
```diff
@@ -6,19 +6,23 @@
...
      *
      * @see https://loophp-collection.readthedocs.io/en/stable/pages/api.html#unfold
      *
-     * @template T
+     * @template U
      *
-     * @param callable(T ...): list<T> $callback
-     * @param T ...$parameters
+     * @param callable((T|U) ...): list<T> $callback
+     * @param array<int, U> $parameters
```
Comment (AlexandruGG) [inline src/Contract/Operation/Unfoldable.php:23]: so does this mean that now in `$parameters` we want to be able to pass values of different types? What would be a good use case for that?

Comment (drupol) [reply src/Contract/Operation/Unfoldable.php:23]: Yes we could potentially do that.

Comment (drupol) [reply src/Contract/Operation/Unfoldable.php]: I gave a second look yesterday and you are perfectly right. I reverted some of the changes.

It's ready again for review.

File: src/Contract/Operation/Unfoldable.php [hunk 2]
```diff
@@ -6,19 +6,23 @@
...
+ */
 interface Unfoldable
 {
     /**
      * Create a collection by yielding from a callback with an initial value.
      *
      * @see https://loophp-collection.readthedocs.io/en/stable/pages/api.html#unfold
      *
-     * @template T
+     * @template U of T
```
Comment (AlexandruGG) [inline src/Contract/Operation/Unfoldable.php:20]: I'm not sure why this is needed - doesn't it work if we leave it like before, just `T` ? Either way I guess it's the same thing

**Review by AlexandruGG — APPROVED** [review]

File: src/Contract/Operation/Unfoldable.php (hunk 2 above)
Comment (drupol) [reply src/Contract/Operation/Unfoldable.php:20]: This is a static function, so, `U` must be defined locally at the method level. And `U` is going to be made of `T`.

**Pull request merged**